library/src/main/java/com/datatorrent/lib/math/RunningAverage.java

```diff
@@ -64,9 +64,9 @@
     @Override
     public void process(Number tuple)
     {
-      double sum = (count * average) + tuple.doubleValue();
-      count++;
-      average = sum / count;
+      double sum = (RunningAverage.this.count * average) + tuple.doubleValue();
```

Comment:
@bhupeshchawda This is a problem in the DefaultInputPort. We should keep count as private and provide protected methods to update the count.